models/models.js

```diff
@@ -21,6 +24,33 @@ exports.getAllArticles = () => {
 exports.selectArticleById = ({ params }) => {
     return db.query("SELECT * FROM articles WHERE article_id=$1;", [params.article_id])
         .then((article) => {
-            return article.rows[0]
+            if (article.rows.length !== 0) {
+                return article.rows[0]
+            }
+            else return Promise.reject({
+                message: "Not Found",
+                status: 404
+            })
+        })
+}
+
+exports.selectCommentsByArticle = ({ params }) => {
+    return db.query("SELECT * FROM comments WHERE article_id=$1 ORDER BY created_at DESC", [params.article_id])

```

Comment:
I like the thinking here, if the response length is 0 we want to check *why* that is the case. My suggestion would be to make that check if id exists call first and if it doesn't exist, throw the error and only if it does exist then proceed to run your standard query.